client/state/themes/reducer.js

```diff
@@ -162,6 +165,34 @@ export function themeRequests( state = {}, action ) {
 }
 
 /**
+ * Returns the updated Jetpack site wpcom theme install requests state after an action has been
+ * dispatched. The state reflects a mapping of site ID, theme ID pairing to a
+ * boolean reflecting whether a request for the theme install is in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function installThemeOnJetpackRequests( state = {}, action ) {
+	switch ( action.type ) {
+		case THEME_INSTALL_ON_JETPACK_REQUEST:
+		case THEME_INSTALL_ON_JETPACK_REQUEST_SUCCESS:
+		case THEME_INSTALL_ON_JETPACK_REQUEST_FAILURE:
+			return Object.assign( {}, state, {
+				[ action.siteId ]: Object.assign( {}, state[ action.siteId ], {
+					[ action.themeId ]: THEME_INSTALL_ON_JETPACK_REQUEST === action.type
```

Comment:
Currently, we would need to know the id of the theme to determine whether a site is currently installing something. Maybe `themeId` should be a value rather than a key?